lib/redix_sentinel.ex

```diff
@@ -317,4 +318,10 @@ defmodule RedixSentinel do
       end
     end)
   end
+
+  defp confirm_role!(conn, role) do
+    [^role | _] = Redix.command!(conn, ["ROLE"])
+    rescue Redix.Error ->
```

Comment:
should we add a check here whether the error is `unknown command?`, because Redix could throw other errors where we don't want to fallback to `INFO`. I see the following test in Redix
```elixir
assert_raise Redix.Error, "ERR unknown command 'NONEXISTENT'", fn ->
Redix.command!(c, ["NONEXISTENT"])
```

Reply:
I was thinking this too and lamented that the `rescue` can't pattern match on the error struct _and_ message (if there is a way to do this, please let me know).

I've updated the code to rescue the `Redix.Error` and then do an `if` on the message. If it matches, I call `INFO replication`, else I re-raise the error. It's pretty ugly but it gives us the level of specificity that we want, namely: **only** act on `Redix.Error`s with a message of "ERR unknown command 'ROLE'".